lib/std/os.zig

```diff
@@ -2107,6 +2107,9 @@ pub fn renameatW(
         .ACCESS_DENIED => return error.AccessDenied,
         .OBJECT_NAME_NOT_FOUND => return error.FileNotFound,
         .OBJECT_PATH_NOT_FOUND => return error.FileNotFound,
+        // The way to handle NOT_SAME_DEVICE is to copy the file
+        // and then delete the source file
+        .NOT_SAME_DEVICE => @panic("TODO handle cross-device rename"),
```

Comment:
The equivalent error on posix is EXDEV, we can match it here:

```suggestion
        .NOT_SAME_DEVICE => return error.RenameAcrossMountPoints,
```

Better to return the error in this low level API and have an additional higher level `renameAllowCopy` (maybe `move` ?)  or something like that implemented in a cross platform manner. Ability to attempt a rename and have it fail if it would require a copy could be useful, and we should not prevent the API user from this ability.